kong/dns/utils.lua

```diff
@@ -142,10 +142,15 @@ local function parse_resolv_conf(path, enable_ipv6)
   resolv.search = resolv.search or (resolv.domain and { resolv.domain })
 
   -- check if timeout is 0s
-  if resolv.options.timeout and resolv.options.timeout <= 0 then
-    log(NOTICE, "A non-positive timeout of ", resolv.options.timeout,
-                "s is configured in resolv.conf. Setting it to 2000ms.")
-    resolv.options.timeout = 2000 -- 2000ms is lua-resty-dns default
+  if resolv.options.timeout then
+    if resolv.options.timeout <= 0 then
+      log(NOTICE, "A non-positive timeout of ", resolv.options.timeout,
+                  "s is configured in resolv.conf. Setting it to 2000ms.")
+      resolv.options.timeout = 2000 -- 2000ms is lua-resty-dns default
+
+    else
+      resolv.options.timeout = resolv.options.timeout * 1000
```

Comment:
Should we add a comment to explain this?

Reply:
Something to just note -- resolv.conf s to ms conversion seems reasonable incase someone doesn't wanna look back at the PR and just see a note in the code.

Reply:
```suggestion
      -- convert resolv.conf timeout from seconds to milliseconds
      resolv.options.timeout = resolv.options.timeout * 1000
```

Reply:
fixed. using comment recommented by gpt with a minor modification

Reply:
lgtm